tests/unit/dom-class.js

```diff
@@ -0,0 +1,840 @@
+define([
+    'intern!object',
+    'intern/chai!assert',
+    '../../dom-class',
+    'dojo/dom-construct',
+    'dojo/sniff'
+
+], function (registerSuite, assert, domClass, domConstruct) {
+
+    var cssClass = 'test-css-class';
+    var moreCssClass = 'test-adding-css-class';
+    var multipleCss = 'existing-css-class';
+    var value = 'the value';
+    var node;
+    var nodeIdIndex = 1;
+
+    function generateId() {
+        return 'nodeid_' + nodeIdIndex++;
+    }
+
+    registerSuite({
+        name: 'dojo/dom-class',
+
+        ".contains": {
+            beforeEach: function () {
+                node = domConstruct.toDom("<div class='" +
+                    cssClass + "'></div>");
+                domConstruct.place(node, document.body);
+            },
+
+            afterEach: function () {
+                domConstruct.destroy(node);
+            },
+
+            "node + class": function () {
+                // when a node and class are passed, then true is returned
+
+                var result = domClass.contains(node, cssClass);
+
+                assert.isTrue(result);
+
+            },
+
+            "string + class": function () {
+                // when a node and class are passed, then true is returned
+
+                var nodeId = generateId();
+                node.setAttribute('id', nodeId);
+
+                var result = domClass.contains(nodeId, cssClass);
+
+                assert.isTrue(result);
+
+            },
+
+            "node + unknown class": function () {
+                // when a node and unknown class are passed, then false is returned
+
+                var result = domClass.contains(node, 'unknown-css-class');
+
+                assert.isFalse(result);
+
+            },
+
+            "string + unknown class": function () {
+                // when a node id and unknown class are passed, then false is returned
+
+                var nodeId = generateId();
+                node.setAttribute('id', nodeId);
+
+                var result = domClass.contains(nodeId, 'unknown-css-class');
+
+                assert.isFalse(result);
+
+            },
+
+            "node w/ multiple classes + class": function () {
+                // when a node and class are passed, then true is returned
+
+                node.setAttribute('class', cssClass + ' ' + multipleCss);
+
+                var result = domClass.contains(node, cssClass) &&
+                             domClass.contains(node, multipleCss);
+
+                assert.isTrue(result);
+
```

Comment:
Not important, but did you know that you can have multiple assert() statements in your function?   So I think most people would write this like:

```
assert.isTrue(domClass.contains(node, cssClass));
assert.isTrue(domClass.contains(node, multipleCss));
```

(I also don't know why you keep setting a result variable that you only use once.)